src/controllers/user-controller.js

```diff
@@ -1,43 +1,59 @@
-import User from '../models/user-model';
+import serviceContainer from "../containers/di-container";
+
+const userService = serviceContainer.userService;
 
 exports.addUser = (request, response) => {
-    const {login, password, age} = request.body;
+    const userDTO = request.body;
 
-    const user = new User(login, password, age);
-    user.save();
+    const user = userService.add(userDTO);
 
-    response.status(201).json(user);
+    user.then(user => {
```

Comment:
lets use async/await syntax everywhere when dealing with asynchronous code. It is more convinient and modern way of working with async actions:
```
export const create = async (req, res) => {
    const user = await userService.create(req.body);
    res.status(201).send(user)
};
```

One major point - we MUST handle asynchronous errors.
Handling async errors is a bit tricky in express. By default each sync error is catched by express error handler, but when we deal with async stuff this error middleware can't catch errors (this is because all async actions are executed only after all synchronous code is finished - so current stack is empty).
There are some possible options (I'll describe only applicable to **async/await** syntax):
1) In each router function wrap your async code with try-catch, but this approach is not so beautiful as you may see
2) You can create your custom middleware for handling async errors
3) The easiest option is two use ready solution (which actually implements the 2nd approach) as an eaxmple:  https://www.npmjs.com/package/express-async-errors

This comment is about all the methods with async operations